### PR TITLE
Core/Scripts: Implement Boss Coren Direbrew

### PR DESCRIPTION
**Pull request by sirikfoll**
**Changes proposed:**
-  Implement Coren DIrebrew fight

Testing and feedback will be deeply appreciated :)

**Target branch(es):** 3.3.5/master

**Tests performed:** Built and tested with a single player, more testing is advised

**Known issues and TODO list:**
- [x] Coren should walk before the combat, I added the waypoints to the Db, but he still doesn't walk.

### CODE REVIEW

File: sql/updates/world/3.3.5/9999_99_99_Coren_world.sql [hunk 1]
```diff
@@ -0,0 +1,72 @@
+-- Coren Direbrew Implementation
+SET @CGUID := 600000;
+DELETE FROM `creature` WHERE `id`=23872 OR `guid`=@CGUID;
+INSERT INTO `creature` (`guid`, `id`, `map`, `zoneId`, `areaId`, `spawnMask`, `phaseMask`, `modelid`, `equipment_id`, `position_x`, `position_y`, `position_z`, `orientation`, `spawntimesecs`, `spawndist`, `currentwaypoint`, `curhealth`, `curmana`, `MovementType`, `npcflag`, `unit_flags`, `dynamicflags`, `VerifiedBuild`) VALUES
+(@CGUID, 23872, 230, 0, 0, 1, 1, 0, 1, 891.8394, -129.1829, -49.65985, 5.253441, 300, 0, 0, 0, 0, 0, 0, 0, 0, 0);
```
Comment (Killyana) [inline sql/updates/world/3.3.5/9999_99_99_Coren_world.sql]: `MovementType`=2

Comment (sirikfoll) [inline sql/updates/world/3.3.5/9999_99_99_Coren_world.sql]: Now he walks, thanks :D

**Comment by sirikfoll** [comment]
I have never done this boss in retail, everything was done based in sniffs and videos, if anyone note something wrong or that can be improved, please tell me.

**Comment by Kittnz** [comment]
Can we queue for this one?

**Comment by sirikfoll** [comment]
yes, through dungeon finder, `.event start 24`

**Comment by Keader** [comment]
Looks good to me

**Comment by Exodius** [comment]
Shouldn't there be 2-3 creatures near him that also belong to this event? Or is that 7.x only?

**Comment by tkrokli** [comment]
@Exodius : is this 2008 YT video useful as "proof" that there used to be adds in TBC / WotLK too?
https://www.youtube.com/watch?v=OkHVeBBVCyA (link found on wowwiki & wowpedia)

**Comment by sirikfoll** [comment]
The npcs are there, unlike Coren, they aren't permanent spawn, they are summoned here https://github.com/TrinityCore/TrinityCore/pull/18137/files#diff-6561d0dd9e8fd37ddc7f2c967a8b2d40R151

**Comment by Exodius** [comment]
@tkrokli @sirikfoll I don't know. I was asking. I understand that temp-summons are usually controlled by "main guy" script and I don't know whether or not they existed during WotLK period... I didn't play retail that much during WotLK period. I do know that, in this year's event, there are 2 or 3 NPCs, standing in front of him while he is walking around and speaking. When you Insult and engage him, they start fighting you as well. They aren't summoned after the fight starts, they are spawned before the fight begins... That's what I was asking, actually.

Did NPCs, that are spawned before the fight begins at this year's (retail) brewfest, exist during WotLK brewfest? Were they spawned before you engaged him or summoned after? If it's the former, they should be included in `game_event_creature` table as well.

**Comment by sirikfoll** [comment]
@Exodius  They already existed back in wotlk.
Did you tested this PR? Everything you said is already implemented, and, at least for me, working ^^

**Comment by sirikfoll** [comment]
Oh, there's another issue to be solved..
The GameObject 188509, is one of those traps, that doesn't show any info in sniffs, so i supposed that the data in gameobject_template is guessed, the problem is with the DisplayId, it needs to be invisible, currently it is visible. What displayId should be used in this case?
@Killyana  @Kittnz  Any ideas?

**Comment by Kittnz** [comment]
the infernal or imp model i guess they are invisible, have you checked sniffed?

**Comment by Killyana** [comment]
To make spell focus invisible we used this: https://github.com/TrinityCore/TrinityCore/commit/60bcd6c98cb403e357631c0a666ee8eea753c3e5
But this gob is a trap and  data8: serverOnly= 0, The model is generic supposed to be invisible to players if the gob is serverOnly.

**Comment by sirikfoll** [comment]
@Vincent-Michael  Do you remeber if the values added in the commit bellow were guessed?https://github.com/TrinityCore/TrinityCore/commit/2b9b2a7388703c6424aa8a9f00b10bffbc5fdc96

Because the sniffs doesn't show the template info for this object, just that it cast spell and despawns right after, and i am pretty sure that the values are wrong, but the WDBVerified is set to 17371, so I can't confirm or edit them :/

**Review by Keader — APPROVED** [review]
looks good to me

**Pull request merged**